core/common/src/main/java/org/sopt/official/common/config/remoteconfig/SoptRemoteConfigImpl.kt

```diff
@@ -0,0 +1,20 @@
+package org.sopt.official.common.config.remoteconfig
+
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import javax.inject.Inject
+import kotlinx.coroutines.tasks.await
+import kotlinx.serialization.json.Json
+
+class SoptRemoteConfigImpl @Inject constructor(
+    private val json: Json,
+    private val remoteConfig: FirebaseRemoteConfig
+) : SoptRemoteConfig {
+    override suspend fun getVersionConfig(): Result<UpdateConfigModel> = runCatching {
+        remoteConfig.fetchAndActivate().await()
+        val versionConfig = remoteConfig.getString(UPDATE_CONFIG)
+        json.decodeFromString<UpdateConfigModel>(versionConfig)
+    }.fold(
+        onSuccess = { Result.success(it) },
+        onFailure = { Result.failure(it) }
+    )
```

Comment:
```suggestion
    override suspend fun getVersionConfig(): Result<UpdateConfigModel> = runCatching {
        remoteConfig.fetchAndActivate().await()
        val versionConfig = remoteConfig.getString(UPDATE_CONFIG)
        json.decodeFromString<UpdateConfigModel>(versionConfig)
    }
```

성공값, 실패값만 넘겨주는 거면 fold는 없애도 될 것 같아요